Core/Src/main.c

```diff
@@ -1,181 +1,252 @@
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+
+  // Initialize SPI1
+  SPI_HandleTypeDef hspi1;
+  HAL_SPI_Init(&hspi1);
+  hspi1.Init.Mode = SPI_MODE_MASTER;
+  hspi1.Init.Direction = SPI_DIRECTION_2LINES;
+  hspi1.Init.DataSize = SPI_DATASIZE_8BIT;
+  hspi1.Init.CLKPolarity = SPI_POLARITY_LOW;
+  hspi1.Init.CLKPhase = SPI_PHASE_1EDGE;
+  hspi1.Init.NSS = SPI_NSS_SOFT;
+  hspi1.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_16;
+  hspi1.Init.FirstBit = SPI_FIRSTBIT_MSB;
+  hspi1.Init.TIMode = SPI_TIMODE_DISABLE;
+  hspi1.Init.CRCCalculation = SPI_CRCCALCULATION_DISABLE;
+
+  //Initialize TIM1
+  TIM_HandleTypeDef htim1;
+  HAL_TIM_PWM_Init(&htim1);
+  htim1.Init.Prescaler = WARG_TIM1_PRESCALER;
+  htim1.Init.CounterMode = TIM_COUNTERMODE_UP;
+  htim1.Init.Period = WARG_TIM1_COUNTER_PERIOD;
+  htim1.Init.ClockDivision = TIM_CLOCKDIVISION_DIV1; // No division?
+  htim1.Init.RepetitionCounter = 0;
+  TIM_OC_InitTypeDef octim1;
+  octim1.OCMode = TIM_OCMODE_PWM1;
+  octim1.Pulse = 0;
+  octim1.OCPolarity = TIM_OCPOLARITY_HIGH;
+  octim1.OCNPolarity = TIM_OCNPOLARITY_HIGH;
+  octim1.OCFastMode = TIM_OCFAST_DISABLE;
+  octim1.OCIdleState = TIM_OCIDLESTATE_RESET;
+  HAL_TIM_PWM_ConfigChannel(&htim1, &octim1, WARG_TIM1_PWM_CHANNEL);
+
+  uint16_t size = 2;
+  uint8_t pTxData[2] = {WARG_ADC_SELECT, 0}; // select ch0, single for ADC
+  uint8_t pRxData[2];
+  uint32_t timeout = 0;
+  uint32_t onCount;
+  uint32_t adcData;
+
+
+  HAL_TIM_PWM_Start(&htim1, WARG_TIM1_PWM_CHANNEL);
+  HAL_GPIO_WritePin(GPIOB, WARG_GPIO_PIN, GPIO_PIN_SET);
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+	  // Set CS line low to begin transmission with ADC
+	  HAL_GPIO_WritePin(GPIOB, WARG_GPIO_PIN, GPIO_PIN_RESET);
+	  HAL_SPI_TransmitReceive(&hspi1, pTxData, pRxData, size, timeout);
+	  // Pull CS line high between conversations
+	  HAL_GPIO_WritePin(GPIOB, WARG_GPIO_PIN, GPIO_PIN_SET);
+	  // Format data from ADC, shift off last 6 garbage bits
+	  adcData = ((pRxData[1] << 8)| pRxData[0]) >> 6;

```

Comment:
This is close but if you look at figure 6-1 again in the datasheet again, it should clear some things up.

Based on the figure, 3 bytes are received, the first byte is garbage, the second byte contains some data and the third byte contains data.

In any case, as mentioned above, you need 3 buffer spaces because you still need to receive the first garbage byte, you will just ignore it when putting together your `adcData` value while stitching together the important bits similar to how you have here.

Reply:
I understand now, fixed. Thanks!